lib/segment/src/index/field_index/mmap_point_to_values.rs

```diff
@@ -0,0 +1,437 @@
+        create_and_ensure_length(&file_name, file_size)?;
+        let mut mmap = open_write_mmap(&file_name)?;
+
+        // fill mmap file data
+        let header = Header {
+            ranges_start: PADDING_SIZE as u64,
+            points_count: points_count as u64,
+        };
+        header
+            .write_to_prefix(mmap.as_mut())
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))?;
+
+        // counter for values offset
+        let mut point_values_offset = header.ranges_start as usize + ranges_size;
+        for (point_id, values) in iter {
+            let start = point_values_offset;
+            let mut values_count = 0;
+            for value in values {
+                values_count += 1;
+                let bytes = mmap.get_mut(point_values_offset..).ok_or_else(|| {
+                    OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE)
+                })?;
+                T::write_to_mmap(value.clone(), bytes).ok_or_else(|| {
+                    OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE)
+                })?;
+                point_values_offset += T::mmaped_size(value);
+            }
+
+            let range = MmapRange {
+                start: start as u64,
+                count: values_count as u64,
+            };
+            mmap.get_mut(
+                header.ranges_start as usize
+                    + point_id as usize * std::mem::size_of::<MmapRange>()..,
+            )
+            .and_then(|bytes| range.write_to_prefix(bytes))
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))?;
+        }
+
+        mmap.flush()?;
+        Ok(Self {
+            file_name,
+            mmap: mmap.make_read_only()?,
+            header,
+            phantom: std::marker::PhantomData,
+        })
+    }
+
+    pub fn open(path: &Path) -> OperationResult<Self> {
+        let file_name = path.join(POINT_TO_VALUES_PATH);
+        let mmap = open_write_mmap(&file_name)?;
+        let header =
+            Header::read_from_prefix(mmap.as_ref()).ok_or(OperationError::InconsistentStorage {
+                description: NOT_ENOUGHT_BYTES_ERROR_MESSAGE.to_owned(),
+            })?;
+
+        Ok(Self {
+            file_name,
+            mmap: mmap.make_read_only()?,
+            header,
+            phantom: std::marker::PhantomData,
+        })
+    }
+
+    fn files(&self) -> Vec<PathBuf> {
+        vec![self.file_name.clone()]
+    }
+
+    pub fn check_values_any(
+        &self,
+        point_id: PointOffsetType,
+        check_fn: impl Fn(T::Referenced<'_>) -> bool,
+    ) -> bool {
+        if point_id < self.header.points_count as PointOffsetType {
+            let range_offset = (self.header.ranges_start as usize)
+                + (point_id as usize) * std::mem::size_of::<MmapRange>();
+            self.mmap
+                .get(range_offset..range_offset + std::mem::size_of::<MmapRange>())
```

Comment:
This logic is duplicated in `check_values_any`, `get_values`, `get_values_count`.

I suggest to add a method like this:
`fn get_range(&self, point_id: PointOffsetType) -> Option<MmapRange>`

Reply:
done